From 4b4cfd70dbcb7b76ce3da0357f15629cb67025fb Mon Sep 17 00:00:00 2001
From: Daniel Wang <wonderfly@google.com>
Date: Thu, 2 Feb 2017 15:29:54 -0800
Subject: [PATCH] systemd: Patch service files

Specifically:
  - Group all services from this package into system-sysdaemons.slice
  - Ensure none of them blocks sshd.service
---
 google_compute_engine_init/systemd/google-accounts-daemon.service   | 2 +-
 google_compute_engine_init/systemd/google-clock-skew-daemon.service | 1 +
 google_compute_engine_init/systemd/google-instance-setup.service    | 6 +++---
 .../systemd/google-ip-forwarding-daemon.service                     | 1 +
 google_compute_engine_init/systemd/google-network-setup.service     | 5 +++--
 google_compute_engine_init/systemd/google-shutdown-scripts.service  | 5 +++--
 google_compute_engine_init/systemd/google-startup-scripts.service   | 5 +++--
 7 files changed, 15 insertions(+), 10 deletions(-)

diff --git a/google_compute_engine_init/systemd/google-accounts-daemon.service b/google_compute_engine_init/systemd/google-accounts-daemon.service
index 125e776..001107c 100644
--- a/google_compute_engine_init/systemd/google-accounts-daemon.service
+++ b/google_compute_engine_init/systemd/google-accounts-daemon.service
@@ -1,12 +1,12 @@
 [Unit]
 Description=Google Compute Engine Accounts Daemon
 After=network.target google-instance-setup.service google-network-setup.service
-Before=sshd.service
 Requires=network.target
 
 [Service]
 Type=simple
 ExecStart=/usr/bin/google_accounts_daemon
+Slice=system-sysdaemons.slice
 
 [Install]
 WantedBy=multi-user.target
diff --git a/google_compute_engine_init/systemd/google-clock-skew-daemon.service b/google_compute_engine_init/systemd/google-clock-skew-daemon.service
index 476abde..4d4b217 100644
--- a/google_compute_engine_init/systemd/google-clock-skew-daemon.service
+++ b/google_compute_engine_init/systemd/google-clock-skew-daemon.service
@@ -6,6 +6,7 @@ Requires=network.target
 [Service]
 Type=simple
 ExecStart=/usr/bin/google_clock_skew_daemon
+Slice=system-sysdaemons.slice
 
 [Install]
 WantedBy=multi-user.target
diff --git a/google_compute_engine_init/systemd/google-instance-setup.service b/google_compute_engine_init/systemd/google-instance-setup.service
index afa639a..6f8f2c5 100644
--- a/google_compute_engine_init/systemd/google-instance-setup.service
+++ b/google_compute_engine_init/systemd/google-instance-setup.service
@@ -1,12 +1,12 @@
 [Unit]
 Description=Google Compute Engine Instance Setup
-After=local-fs.target network-online.target network.target rsyslog.service
-Before=sshd.service
-Wants=local-fs.target network-online.target network.target
+After=network-online.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
 ExecStart=/usr/bin/google_instance_setup
+Slice=system-sysdaemons.slice
 
 [Install]
 WantedBy=sshd.service
diff --git a/google_compute_engine_init/systemd/google-ip-forwarding-daemon.service b/google_compute_engine_init/systemd/google-ip-forwarding-daemon.service
index d3704c6..491a0b1 100644
--- a/google_compute_engine_init/systemd/google-ip-forwarding-daemon.service
+++ b/google_compute_engine_init/systemd/google-ip-forwarding-daemon.service
@@ -6,6 +6,7 @@ Requires=network.target
 [Service]
 Type=simple
 ExecStart=/usr/bin/google_ip_forwarding_daemon
+Slice=system-sysdaemons.slice
 
 [Install]
 WantedBy=multi-user.target
diff --git a/google_compute_engine_init/systemd/google-network-setup.service b/google_compute_engine_init/systemd/google-network-setup.service
index 4400391..abc0544 100644
--- a/google_compute_engine_init/systemd/google-network-setup.service
+++ b/google_compute_engine_init/systemd/google-network-setup.service
@@ -1,13 +1,14 @@
 [Unit]
 Description=Google Compute Engine Network Setup
-After=local-fs.target network-online.target network.target rsyslog.service
+After=network-online.target
 After=google-instance-setup.service
-Wants=local-fs.target network-online.target network.target
+Wants=network-online.target
 
 [Service]
 ExecStart=/usr/bin/google_network_setup
 KillMode=process
 Type=oneshot
+Slice=system-sysdaemons.slice
 
 [Install]
 WantedBy=multi-user.target
diff --git a/google_compute_engine_init/systemd/google-shutdown-scripts.service b/google_compute_engine_init/systemd/google-shutdown-scripts.service
index 3561089..9bad914 100644
--- a/google_compute_engine_init/systemd/google-shutdown-scripts.service
+++ b/google_compute_engine_init/systemd/google-shutdown-scripts.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Google Compute Engine Shutdown Scripts
-After=local-fs.target network-online.target network.target rsyslog.service
+After=network-online.target
 After=google-instance-setup.service google-network-setup.service
-Wants=local-fs.target network-online.target network.target
+Wants=network-online.target
 
 [Service]
 ExecStart=/bin/true
@@ -10,6 +10,7 @@ ExecStop=/usr/bin/google_metadata_script_runner --script-type shutdown
 Type=oneshot
 RemainAfterExit=true
 TimeoutStopSec=0
+Slice=system-sysdaemons.slice
 
 [Install]
 WantedBy=multi-user.target
diff --git a/google_compute_engine_init/systemd/google-startup-scripts.service b/google_compute_engine_init/systemd/google-startup-scripts.service
index 9c04d79..05f4def 100644
--- a/google_compute_engine_init/systemd/google-startup-scripts.service
+++ b/google_compute_engine_init/systemd/google-startup-scripts.service
@@ -1,13 +1,14 @@
 [Unit]
 Description=Google Compute Engine Startup Scripts
-After=local-fs.target network-online.target network.target rsyslog.service
+After=network-online.target
 After=google-instance-setup.service google-network-setup.service
-Wants=local-fs.target network-online.target network.target
+Wants=network-online.target
 
 [Service]
 ExecStart=/usr/bin/google_metadata_script_runner --script-type startup
 KillMode=process
 Type=oneshot
+Slice=system-sysdaemons.slice
 
 [Install]
 WantedBy=multi-user.target
-- 
2.11.0.483.g087da7b7c-goog

